client/dashboard/domains/name-servers/utils.ts

```diff
@@ -0,0 +1,18 @@
+export const WPCOM_DEFAULT_NAMESERVERS_REGEX = /^ns[1-4]\.wordpress\.com$/i;
+
+export const HOSTNAME_REGEX =
+	/^(?=.{1,255}$)(?:[a-zA-Z0-9](?:[a-zA-Z0-9-]{0,61}[a-zA-Z0-9])?\.)+[a-zA-Z]{2,}$/;
+
+export const validateHostname = ( hostname: string ) => {
```

Comment:
Let's use the browser to do the heavy lifting for us: https://stackoverflow.com/a/73860244